src/lib/services/feature-toggle-service-v2.ts

```diff
@@ -374,6 +375,57 @@ class FeatureToggleServiceV2 {
         throw new NotFoundError(`Project with id ${projectId} does not exist`);
     }
 
+    async cloneFeatureToggle(
+        featureName: string,
+        projectId: string,
+        newFeatureName: string,
+        replaceGroupId: boolean = true,
+        userName: string,
+    ): Promise<FeatureToggle> {
+        this.logger.info(
+            `${userName} clones feature toggle ${featureName} to ${newFeatureName}`,
+        );
+        await this.validateName(newFeatureName);
+
+        const cToggle =
+            await this.featureStrategiesStore.getFeatureToggleWithEnvs(
+                featureName,
+            );
+
+        const newToggle = { ...cToggle, name: newFeatureName };
+
+        // Create feature toggle
+        const created = await this.createFeatureToggle(
+            projectId,
+            newToggle,
+            userName,
+        );
+
+        const createStrategies = [];
+        newToggle.environments.forEach((e: IEnvironmentDetail) =>
+            e.strategies.forEach((s: IStrategyConfig) => {
+                if (replaceGroupId && s.parameters.hasOwnProperty('groupId')) {
+                    //@ts-ignore
+                    s.parameters.groupId = newFeatureName;
+                }
+                delete s.id;
+                createStrategies.push(
+                    this.createStrategy(
+                        s,
+                        projectId,
+                        newFeatureName,
+                        userName,
+                        e.name,
+                    ),
+                );
+            }),
+        );
+
+        // Create strategies
```

Comment:
actually, we are waiting for all the create strategies to be completed. The async function was started when we added the call to the createStrategy array.